src/Util/QueryBuilder.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Mongolid\Util;
+
+use MongoDB\BSON\ObjectId;
+use Mongolid\Model\ModelInterface;
+
+class QueryBuilder
+{
+    public static function resolveQuery(int|array|string $query, ModelInterface $model): array
```

Comment:
```suggestion
    public static function resolveQuery(mix $query, ModelInterface $model): array
```
I think we can pass more types than specified here, like `MongoDB\BSON\ObjectId`. What do you think to change it to _mixed_?

Reply:
Change made